include/vcli_serve.h

```diff
@@ -47,6 +47,18 @@ enum cli_flags {
 #include "tbl/cli_flags.h"
 };
 
+#define _FLG_IS(d, f) (d->flags & f)
+
+#define CMD_AUTH(c) _FLG_IS(c, CLI_F_AUTH)
+#define CMD_INTERNAL(c) _FLG_IS(c, CLI_F_INTERNAL)
+#define CMD_SENSITIVE(c) _FLG_IS(c, CLI_F_SENSITIVE)
+#define CMD_DEBUG(c) _FLG_IS(c, CLI_F_DEBUG)
+
+#define CLP_AUTH(c) CMD_AUTH(c->desc)
+#define CLP_INTERNAL(c) CMD_INTERNAL(c->desc)
+#define CLP_SENSITIVE(c) CMD_SENSITIVE(c->desc)
+#define CLP_DEBUG(c) CMD_DEBUG(c->desc)
```

Comment:
We´d probably be better off with just two macros:

```c
#define VCLS_CMD_IS(cmd, FLAG) ((cmd)->flags & CLI_F_##FLAG)
#define VCLS_PROTO_IS(clp, FLAG) VCLS_CMD_IS((clp)->desc, FLAG)
```

And then you use it like this:

```c
if (VCLS_PROTO_IS(AUTH) && !cli->auth)
        continue;
```

This way you don't need to add new macros for new flags.

And I made sure to namespace them in the suggestion above.

Reply:
Fixed in latest revision